core/src/main/scala/chisel3/package.scala

```diff
@@ -234,6 +235,117 @@ package object chisel3 {
+     *  `java.lang.String.format` to format all arguments with that format string. The format string is
+     *  obtained by concatenating all `parts` strings, and performing two transformations:
+     *   1. Let a _formatting position_ be a start of any `parts` string except the first one.
+     *      If a formatting position does not refer to a `%` character (which is assumed to
+     *      start a format specifier), then the string format specifier `%s` is inserted. This means any string with no format specifiers should behave same as p interpolator.
+     *   2. If format specifiers specifically target Chisel types - Like Data/Bits -  specific internal methods are called on that argument (like FirrtlFormat)
+     *   3. If format specifers target regular scala data types - java.lang.String.format method is called with specifier and argument to get the formatted string.
+     *
+     * Yet to document all the corener cases which this code tries to cover - but here are few of them
+     * 1. Allow calling %f or %2.2f (or something similar) on Int type data (Int, Short, Long) - this requires doing an explicit cast of the data before calling String.format
+     * 2. For bits - if no format specifier given (%s i.e) call .toPrintable to pick the default one. For regular data - let the String.format handle canonical case of %s too.
+     * Still testing this code - I am pretty sure this can be refactored and made more efficient / readable. WIP.
+     *
+     * /
+     */
+     */
+    def cf(args: Any*): Printable = {
+      sc.checkLengths(args) // Enforce sc.parts.size == pargs.size + 1
+      val parts = sc.parts.map(StringContext.treatEscapes)
+
+      // The 1st part is assumed never to contain a format specifier.
+      // If the 1st part of a string is an argument - then the 1st part will be an empty String.
+      // So we need to parse parts following the 1st one to get the format specifiers if any
+      val partsAfterFirst = parts.slice(1, parts.size)
+
+      // Align parts to their potential specifiers
+      val partsAndSpecifierSeq = partsAfterFirst.zip(args).map {
+        case (part, arg) => {
+
+          // Check if part starts with a format specifier (with % - disambiguate with literal % checking the next character if needed to be %)
+          // In the case of %f specifier there is a chance that we need more information - so capture till the 1st letter (a-zA-Z).
+          // Example cf"This is $val%2.2f here" - parts - Seq("This is ","%2.2f here") - the format specifier here is %2.2f.
+          val idx_of_fmt_str =
+            if (
+              !part.isEmpty() && part.charAt(0) == '%' && (part.size == 1 || (part.size >= 2 && part.charAt(1) != '%'))
+            ) part.indexWhere { _.isLetter }
+            else -1
+
+          // If no format specifier - pick default - %s
+
+          val fmt = if (idx_of_fmt_str >= 0) part.substring(0, idx_of_fmt_str + 1) else "%s"
```

Comment:
I don't think it's right to use `%s` as some "default" format-specifier. `%s` means "format as String" which is actually not the same thing as "format as Printable" which is how it seems to be used below. I think you should make `fmt` be `Option[String]` where `None` is the default (no specifier provided) and only the `None` case is `.toPrintable`.

Reply:
Agreed - picked %s as default just based on what is done for normal scala f interpolator.

Reply:
Converted to Option[String] and used .toPrintable for Printable and %s for regular types.